Update and expand Build System feature module documentation (#887)
spatialos/gdk-for-unity: docs/modules/build-system/overview.md

> # Build System Feature Module

Reviewer: probably shouldn't capitalize it all? same everywhere else in the doc

> This feature module contains a default implementation and tooling in the Unity Editor for building your GDK for Unity workers.

Reviewer: ```suggestion
This feature module provides tooling for building your GDK for Unity workers inside the Unity Editor.
```

at least I think location needs to be at the end of the sentence

> **Note:** The Build System Feature Module does not have an integration for building non-Unity workers at this time.

## Installation

> ### 1. Add the package

Reviewer: if there is only one step, should we enumerate it? ;)

Author: Mostly just consistency across the feature modules but can just remove it
